runtime/lua/vim/diagnostic.lua

```diff
@@ -12,6 +12,9 @@ M.severity = {
 
 vim.tbl_add_reverse_lookup(M.severity)
 
+api.nvim_set_hl(0, 'DiagnosticDeprecated', { link = '@lsp.mod.deprecated', default = true })
```

Comment:
Can we instead link to a generic `Deprecated` highlight group so that `vim.diagnostic` remains completely ignorant of LSP/semantic tokens? That highlight group could in turn link to `@lsp.mod.deprecated` (or vice versa).

Reply:
Yeah, probably not worth having this for 0.9.

Reply:
Sure, but I'd still consider the `Diagnostic` groups more fundamental than the LSP mod groups -- which aren't even set by default, so this link is likely to be dangling?

Reply:
I'm suggesting deferring this PR until 0.10 and gating it on defining default groups. What are you suggesting?

Reply:
I'm suggesting including this in 0.9 and just defining a `DiagnosticDeprecated` highlight group in `highlight_groups.c` (ignoring the LSP group for now).

Reply:
No strong opinion either way; as long as the name stays, any later change to how it's set or what it defaults to will not be breaking, so I don't see a good reason to delay this.